### ISSUE TITLE
Feature request: transparent proxying

### ISSUE THREAD

**Issue by patrakov** [paultag/sniff#7]
It would be nice if the proxied-to service could see the client's IP address.

**Comment by paultag** [comment]
Yeah, agree. That'd involve handing the fd off to the remote; which may actually not be possible if it's not on localhost, and at the least, the handoff code should be rewritten.

Might be a bit awkward as the code exists currently (since you need to do a non-destructive read on the fd, or stuff data back into the fd). However, doing this also has a neat benefit of not requiring the daemon to be online to let traffic continue to flow.

So, love the idea, huge +1, but I'm not sure I know go socket internals well enough to do the fd handoff safely. Thoughts?

**Comment by patrakov** [comment]
I was thinking more about non-local binds and iptables tricks, not fd passing. Something like what stunnel does with "transparent = source", see the local_bind() function in their client.c.

**Comment by paultag** [comment]
oh yeah, that sounds like a good idea. Thanks! Someone should feel free to PR this, otherwise I'll see if I can pick this up when I have time :+1:
